dask_kubernetes/operator/controller/controller.py

```diff
@@ -69,15 +77,16 @@ def _get_labels(meta):
 
 
 def build_scheduler_deployment_spec(
-    cluster_name, namespace, pod_spec, annotations, labels
-):
-    labels.update(
-        **{
-            "dask.org/cluster-name": cluster_name,
-            "dask.org/component": "scheduler",
-            "sidecar.istio.io/inject": "false",
-        }
-    )
+    cluster_name: str,
+    pod_spec: kopf.Spec,
+    annotations: kopf.Annotations,
+    labels: kopf.Labels,
+) -> dict[str, Any]:
+    labels = dict(labels) | {
```

Comment:
This is exciting! I hadn't realised union operators were in 3.9, which is our minimum version!

Reply:
The change here actually also removes some remaining in-place modifications of mutable variables :)